### PR TITLE
fix libmysqlclient compilation for cpp 17

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **libmysqlclient/8.0.25**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
fixes https://github.com/conan-io/conan-center-index/pull/5677

**Comment by AndreyMlashkin** [comment]
My changes only affect libmysqlclient/8.0.25, has someone a clue, why libmysqlclient/8.0.17 is failing now?
new conan version bug? sporadic issue?

**Comment by ericLemanissier** [comment]
I have no idea, the build passed 6 days ago, and nothing changed on the recipe or its requirements since then. This is a mystery to me. Maybe just restart the build by closing/reopening the PR?

**Comment by AndreyMlashkin** [comment]
still the same problem with v8.0.17.
Could someone investigate, what's wrong with the CI?

File: recipes/libmysqlclient/all/conandata.yml [hunk 1]
```diff
@@ -13,3 +13,7 @@ patches:
       patch_file: "patches/0002-dont-install-static-libraries+fix-mysql-config.patch"
     - base_path: "source_subfolder"
       patch_file: "patches/0003-msvc-install-no-pdb.patch"
+patches:
```
Comment (ericLemanissier) [inline recipes/libmysqlclient/all/conandata.yml:16]:
```suggestion
```

**Review by ericLemanissier — CHANGES_REQUESTED** [review]
@AndreyMlashkin I found the problem: by declaring a second "patches" key in conandata.yml, it removed all the patches of 8.0.17

**Review by ericLemanissier — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
@SpaceIm @prince-chrismc
please, review

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**